tests/integration/relations/test_db.py

```diff
@@ -178,10 +179,9 @@ async def test_extensions_blocking(ops_test: OpsTest) -> None:
     await ops_test.model.add_relation(f"{PGB}:db", f"{CLIENT_APP_NAME}:db")
 
     logger.info("Wait for PGB to block due to extensions")
-    await ops_test.model.wait_for_idle(apps=[PGB], status="blocked", timeout=1000)
-    assert (
-        ops_test.model.applications[PGB].units[0].workload_status_message
-        == EXTENSIONS_BLOCKING_MESSAGE
+    leader_unit = await get_leader_unit(ops_test, PGB)
+    await ops_test.model.block_until(
+        lambda: leader_unit.workload_status_message == EXTENSIONS_BLOCKING_MESSAGE, timeout=1000
```

Comment:
This is the main divergence I found from the previous behaviour. Since only the leader evaluates the the legacy `_on_relation_joined()` now, only the leader will block.